config/routes.rb

```diff
@@ -100,6 +100,8 @@
     resources :programming_groups, only: %i[new create]
```

Comment:
We might also think to link the programming groups here, in which case they should be pre-filtered for the given exercise. You can determine whether we are in a scoped environment with the `params[:exercise_id]` (and, if called, the `@exercise` variable). In that case, the Ransack part (in the controller and the view) can be hidden.